bitcoin/src/address/mod.rs

```diff
@@ -115,14 +135,27 @@ pub trait NetworkValidation: sealed::NetworkValidation + Sync + Send + Sized + U
     const IS_CHECKED: bool;
 }
 
+/// Marker trait for `FromStr` and `serde::Deserialize`.
+///
+/// This allows users to use `V: NetworkValidation` in conjunction with derives. Is only ever
+/// implemented for `NetworkUnchecked`.
+pub trait NetworkValidationUnchecked:
+    NetworkValidation + sealed::NetworkValidationUnchecked + Sync + Send + Sized + Unpin
+{
+}
+
 /// Marker that address's network has been successfully validated. See section [*Parsing addresses*](Address#parsing-addresses)
 /// on [`Address`] for details.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
```

Comment:
This looks very wrong. The trait impls should be on `Address`, not here. The values cannot be deserialized anyway since they don't exist.

Reply:
No, I think these are needed because of Rust's idiotic `derive` semantics. In order to derive these for `Address<T>` they need to be derived for `T`, even if `T` is not constructible.

Reply:
IIRC `serde` has a way to override those bounds. I'm pretty sure I've had this problem many years ago and it was solvable. But I believe it *is* true for non-serde derives. I remember complaining about it but the discussion stalled because the solution is not obvious. A naive one would leak internal types in public API in some cases.

Reply:
#3974

Reply:
Noted, but its not that important right now right? `address` is not the priority for stabalising so we can punt this down the park.